dot/rpc/modules/childstate_integration_test.go

```diff
@@ -0,0 +1,298 @@
+//go:build integration
+// +build integration
+
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+package modules
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ChainSafe/chaindb"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChildStateGetKeys(t *testing.T) {
+	childStateModule, currBlockHash := setupChildStateStorage(t)
+
+	req := &GetKeysRequest{
+		Key:    []byte(":child_storage_key"),
+		Prefix: []byte{},
+		Hash:   nil,
+	}
+
+	res := make([]string, 0)
+	err := childStateModule.GetKeys(nil, req, &res)
+	require.NoError(t, err)
+	require.Len(t, res, 3)
+
+	for _, r := range res {
+		b, dErr := common.HexToBytes(r)
+		require.NoError(t, dErr)
+		require.Contains(t, []string{
+			":child_first", ":child_second", ":another_child",
+		}, string(b))
```

Comment:
That's risky since if `b` is always one of them, that will pass.

Instead change that to a set `map[string]struct{}{":child_first": {}, ...}` which you check with

```go
key := string(b)
_, ok := set[key]
assert.True(t, ok)
delete(set, key)
```